pauperformance_bot/entity/deck/playable.py

```diff
@@ -36,23 +37,38 @@ def __eq__(self, other):
 
 
 class PlayableDeck:
-    MAIN_DECK_AMOUNT = 60
-    SIDEBOARD_AMOUNT = 15
+    MAINBOARD_MIN_AMOUNT = 60
+    SIDEBOARD_MAX_AMOUNT = 15
+    MAX_NON_LAND_QUANTITY = 4
 
     @classmethod
-    def _validate_board(cls, board, amount):
-        if reduce(lambda t, s: t + s.quantity, board, 0) != amount:
-            raise ValueError(f"Input board does not contain {amount} cards")
+    def validate_boards(cls, mainboard, sideboard) -> Tuple[bool, List[str]]:
+        errors = []
+
+        main_amt = reduce(lambda t, s: t + s.quantity, mainboard, 0)
```

Comment:
Humans tend to prefer `sum` in these cases.
Something like `sum(c.quantity for c in mainboard)` is probably enough.
`reduce` is overkill when the reducing `fun` is `+`.
In fact, one may argue that you just reimplemented Python's `sum` via `reduce`. :)

Reply:
Ahahahah